src/internal_events/process.rs

```diff
@@ -17,6 +18,15 @@ impl InternalEvent for VectorStarted {
             arch = built_info::TARGET_ARCH,
             build_id = built_info::VECTOR_BUILD_DESC.unwrap_or("none"),
         );
+        gauge!(
+            "vector_info",
+            1.0,
+            "debug" => built_info::DEBUG,
+            "version" => built_info::PKG_VERSION,
+            "rust_version" => built_info::RUST_VERSION,
+            "arch" => built_info::TARGET_ARCH,
+            "build_id" => built_info::VECTOR_BUILD_DESC.unwrap_or("none")
```

Comment:
It seems like this is always `none`, even in release builds. Do you know when this value would be set? We may want to avoid adding it here if it never varies.

Reply:
On actual releases I think we have it set:

```shell
❯ vector --version
vector 0.24.0 (x86_64-apple-darwin 43267b9 2022-08-30)
```

`43267b9` is `build_id` IIRC.

Reply:
Ah, good catch. I forgot I manually compiled my release build 😄 . Maybe we should call this tag `sha` then?

Reply:
If we do, we should change it for the rest of the InternalEvent but 👍

Reply:
Yes, I also think that `id` is not a good label name. Indeed `sha` or `revision` might be better.
See https://www.robustperception.io/exposing-the-software-version-to-prometheus/

Reply:
Ah, normally when there are no values for a label/tag, you would use an empty string, not `none`.

Reply:
Changed ```build_id``` to ```revision```